client/asset/btc/btc.go

```diff
@@ -654,6 +654,29 @@ func (d *Driver) Info() *asset.WalletInfo {
 	return WalletInfo
 }
 
+type CustomSPVWalletConstructor func(settings map[string]string, params *chaincfg.Params) (BTCWallet, error)
+
+// customSPVWalletConstructors are functions for setting up custom
+// implementations of the BTCWallet interface that may be used by the
+// ExchangeWalletSPV instead of the default spv implementation.
+var customSPVWalletConstructors = map[string]CustomSPVWalletConstructor{}
+
+// RegisterCustomSPVWallet registers a function that should be used in creating
+// a BTCWallet implementation that the ExchangeWalletSPV will use in place of
+// the default spv wallet implementation. External consumers can use this
+// function to provide alternative BTCWallet implementations, and must do so
+// before attempting to create an ExchangeWalletSPV instance of this type.
+func RegisterCustomSPVWallet(constructor CustomSPVWalletConstructor, def *asset.WalletDefinition) error {
+	for _, availableWallets := range WalletInfo.AvailableWallets {
+		if def.Type == availableWallets.Type {
+			return fmt.Errorf("(%q): %w", def.Type, asset.ErrWalletTypeAlreadyRegistered)
+		}
```

Comment:
I know you're just adopting the existing pattern from dcr, but this should really be a `panic`, I think.

Reply:
I think a panic is appropriate. The consumer can use something like an atomic.Bool to track if the custom type has been registered and avoid repeat registration attempts.